### ISSUE TITLE
Converting float64 to float32 rounds off data in hdu object

### ISSUE THREAD

**Issue by indiajoe** [astropy/astropy#6845]
Hi,
Why do we round off the data to integers before changing the data type in this line?
https://github.com/astropy/astropy/blob/5f88a0ede0d03cf6329d9d5e0971d540275a8b55/astropy/io/fits/hdu/image.py#L549

To save hardisk space, I wanted to convert my float64 fits images to float32.
I used the `hdulist[0].scale('float32')` method to do that.
However, instead of changing float64 to float32, it rounded off to nearest integer.

**Comment by pllim** [comment]
`git blame` pointed this code all the way back to PyFITS, which was added to `astropy` back in #102.

🙏 @embray , could you please comment on this, if you still remember? 🙏

**Comment by pllim** [comment]
In the meantime, is such a workaround feasible for you?

```python
>>> hdulist.info()  # Let's say this is your original HDU list
Filename: (No file associated with this HDUList)
No.    Name      Ver    Type      Cards   Dimensions   Format
  0                1 ImageHDU         7   (10, 10)   float64  
>>> hdu = hdulist[0]
>>> hdu.data = hdu.data.astype('float32')
>>> hdulist = fits.HDUList([hdu])
>>> hdulist.info()
Filename: (No file associated with this HDUList)
No.    Name      Ver    Type      Cards   Dimensions   Format
  0                1 ImageHDU         7   (10, 10)   float32
```

**Comment by indiajoe** [comment]
@pllim , Yes. That is what I did now as a workaround.
used the `.data.astype()`, and it worked as expected.

**Comment by indiajoe** [comment]
I guess the `np.around()` might have been added for `float` to `int` conversions.
`np.array(4.99999,dtype=np.int)` would give 4 instead of 5 if we don't do a `np.around()`

So, I think we might need to keep the `np.around()` call if `_type` is some `int`, and not do it if it is a `float`

**Comment by saimn** [comment]
From what I understand the `.scale` method should only be used to scale to (unsigned) int. (From the FITS standard, *"Use of the BSCALE and BZERO keywords is not recommended"* for floats).

For a float64->float32 conversion, converting the data directly with `.astype` is the good way.

**Comment by indiajoe** [comment]
Oh.. that makes sense. So maybe we should raise some Warning or exception when we provide any 'float' as the dtype.

**Comment by saimn** [comment]
> So maybe we should raise some Warning or exception when we provide any 'float' as the dtype.

Yes, I think so. It's not strictly forbidden with FITS though.

**Comment by saimn** [comment]
Looking at the code this was supported at some point so let's fix it: #17458
